### PR TITLE
UITOOL-810/bump to react 18

### PR DESCRIPTION
**Pull request by aiheon**
### Proposed Changes

UITOOL-810 part 4

Actually bumping React to version 18 😎 
Also, bumping RTL and friends to latest + fixing/removing/skipping uts

I did not bother too much in fixing them since the lib is deprecated 🙃 

My next step is to actually REMOVE enzyme and all the uts using it 

### Potential Breaking Changes

React 18 is needed as peerDependecies

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [X] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by gdostie — COMMENTED** [review]
There seem to be a lot of stuff in this PR, is it possible that some of these changes are already on master? If it is the case it would be worth merging master in next first to clear the diff up

**Comment by aiheon** [comment]
> There seem to be a lot of stuff in this PR, is it possible that some of these changes are already on master? If it is the case it would be worth merging master in next first to clear the diff up

https://github.com/coveo/plasma/pull/2849 👀

File: packages/react/src/components/actions/ActionBar.tsx [hunk 1]
```diff
@@ -131,7 +131,7 @@ export class ActionBar extends PureComponent<
         return classNames(this.props.removeDefaultContainerClasses ? '' : ActionBar.defaultClasses);
     }
 
-    private Actions: FunctionComponent = () => {
+    private Actions: FunctionComponent<React.PropsWithChildren<unknown>> = () => {
```
[the remark this answers is not in the document]
Comment (aiheon) [reply packages/react/src/components/actions/ActionBar.tsx:134]: I can do a search and replace now that the codemod went through everything 🤔 👍

[the remark this answers is not in the document]
Comment (GermainBergeron) [reply packages/react/src/components/actions/ActionBar.tsx:134]: As discussed on Slack, we should only use `PropsWithChildren` when we actually allow children

[the remark this answers is not in the document]
Comment (aiheon) [reply packages/react/src/components/actions/ActionBar.tsx]: 256 to 194 files changed!

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**